typescript/src/marcos/Diamond.ts

```diff
@@ -0,0 +1,28 @@
+import { parse } from "path";
+
+export class Diamond {
+  constructor() {}
+
+  print(letter) {
+
+    if (letter === "A") {
+      return "A";
+    }
+
+    let result = " ".repeat(letter.charCodeAt(0) - 65)  + "A/n";
+
+    for (let i = 66; i <= letter.charCodeAt(0); i++) {
+      result +=
+        " ".repeat(letter.charCodeAt(0) - i) + String.fromCharCode(i) + " ".repeat((i-66)*2+1) + String.fromCharCode(i) + "/n";
+    }
+
+    for (let i = letter.charCodeAt(0)-1; i >= 66; i--) {
+      result +=
+        " ".repeat(letter.charCodeAt(0) - i) + String.fromCharCode(i) + " ".repeat((i-66)*2+1) + String.fromCharCode(i) + "/n";
+    }
```

Comment:
Muy bueno lo que hiciste con Diamond. No se si notaste que tuviste que hacer un trabajo grande de lógica en un solo paso, para eso hay bastante discución acerca de cómo encararlo, el approach de test de propiedades suena muy interesante. Esta otra solución también es piola: https://claysnow.co.uk/recycling-tests-in-tdd/ y tiene una "moraleja" para la práctica diaria:

> By recycling the second test multiple times, each time modifying the expected output to become closer to what we actual want, we’ve allowed ourselves to be guided gradually to a solution.
> 
> When working on user stories, I often encourage teams to deliver low-fidelity slices early, and this is a similar strategy, but applied at a lower level. It won’t work in all cases, but if you find that the next refactor is harder than you’d like, think about trying to get there in smaller steps.